src/component/utils/exploration/__tests__/DraftTreeAdapter-test.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails oncall+ui_infra
+ * @format
+ */
+
+'use strict';
+
+jest.disableAutomock();
+
+const DraftTreeAdapter = require('DraftTreeAdapter');
+
+const assertFromRawTreeStateToRawState = rawState => {
+  expect(DraftTreeAdapter.fromRawTreeStateToRawState(rawState)).toMatchSnapshot();
+};
+
+const assertFromRawStateToRawTreeState = rawState => {
+  expect(
+    DraftTreeAdapter.fromRawStateToRawTreeState(rawState),
+  ).toMatchSnapshot();
+};
+
+test('must be able to convert from tree raw state with only root blocks to raw state', () => {
+  const rawState = {
+    blocks: [
+      {
+        key: 'A',
+        text: 'Alpha',
+        children: [],
+      },
+      {
+        key: 'B',
+        text: 'Beta',
+        children: [],
+      },
+    ],
+    entityMap: {},
+  };
+
+  assertFromRawTreeStateToRawState(rawState);
+});
+
+test('must be able to convert from tree raw state with nested blocks to raw state', () => {
+  const rawState = {
+    blocks: [
+      {
+        key: 'A',
+        type: 'blockquote',
+        text: '',
+        children: [
+          {
+            key: 'B',
+            text: 'Beta',
+            type: 'header-one',
+            children: [],
+          },
+          {
+            key: 'C',
+            text: 'Charlie',
+            type: 'header-two',
+            children: [],
+          },
+        ],
+      },
+    ],
+    entityMap: {},
+  };
+
+  assertFromRawTreeStateToRawState(rawState);
+});
+
+test('must be able to convert from tree raw state with nested list blocks to raw state preserving lists depth', () => {
+  const rawState = {
+    blocks: [
```

Comment:
The snapshot output for this has 'depth' of 0 for all blocks. Should the depth reflect the nesting?

Reply:
This is definitly a bug, must have introduced it while refactoring. Thank you for spotting this!